app/pages/settings/email.spec.js

```diff
@@ -0,0 +1,185 @@
+  display_name: 'A test project',
+  title: 'A test project'
+};
+
+const anotherProject = {
+  display_name: 'Another test project',
+  title: 'Another test project'
+};
+
+const preferences = [
+  {
+    email_communication: true,
+    get() {
+      return Promise.resolve(project);
+    },
+    getMeta() {
+      return {};
+    }
+  },
+  {
+    email_communication: false,
+    get() {
+      return Promise.resolve(anotherProject);
+    },
+    getMeta() {
+      return {};
+    }
+  }
+];
+
+const user = {
+  email: 'An email address',
+  beta_email_communication: false,
+  global_email_communication: true,
+  project_email_communication: true,
+  get() {
+    return Promise.resolve(preferences);
+  }
+};
+
+describe('EmailSettings', () => {
+  const wrapper = mount(<EmailSettings user={user} />);
+
+  beforeEach(() => wrapper.update());
+
+  it('renders the email address', () => {
+    const email = wrapper.find('input[name="email"]');
+    assert.equal(email.prop('value'), user.email);
+  });
+
+  it('shows global email preference correctly', () => {
+    const projectEmail = wrapper.find('input[name="global_email_communication"]');
+    assert.equal(projectEmail.prop('checked'), user.global_email_communication);
+  });
+
+  it('shows project email preference correctly', () => {
+    const projectEmail = wrapper.find('input[name="project_email_communication"]');
+    assert.equal(projectEmail.prop('checked'), user.project_email_communication);
+  });
+
+  it('shows beta email preference correctly', () => {
+    const betaEmail = wrapper.find('input[name="beta_email_communication"]');
+    assert.equal(betaEmail.prop('checked'), user.beta_email_communication);
+  });
+
+  describe('project listing', () => {
+    let projects;
+
+    beforeEach(() => {
+      projects = wrapper.update().find('table').last().find('tbody tr');
+    });
+
+    it('lists two projects (plus pagination)', () => {
+      assert.equal(projects.length, 3);
+    });
+
+    it('shows the first project email input correctly', () => {
+      const email = projects.first().find('td').first();
+      assert.equal(email.find('input').prop('checked'), preferences[0].email_communication);
+    });
+
+    it('shows the first project name correctly', () => {
+      const name = projects.first().find('td').last();
+      assert.equal(name.text(), project.display_name);
+    });
+
+    it('shows the second project email input correctly', () => {
+      const email = projects.at(1).find('td').first();
+      assert.equal(email.find('input').prop('checked'), preferences[1].email_communication);
+    });
+
+    it('shows the second project name correctly', () => {
+      const name = projects.at(1).find('td').last();
+      assert.equal(name.text(), anotherProject.display_name);
+    });
+  });
+
+  describe('Talk email preferences', () => {
+
+    subscriptionPreferences.forEach((preference) => {
+      it('lists `${preference.category} preferences correctly', () => {
```

Comment:
I think this should be ``lists ${preference.category} preferences correctly`,`. Tests still fail though.

Reply:
Good catch! Fixed now.